recipes/Text_to_Shell/Text_to_Shell.ipynb

```diff
@@ -124,28 +295,83 @@
    "source": [
     "## Adding a System Prompt\n",
     "\n",
-    "A system prompt can be used to provide additional instructions and clarity or context for the task. Here we let the model know what we expect from it."
+    "A system prompt can be used to provide additional instructions and clarity or context for the task. Here we let the model know what we expect from it.\n",
+    "\n",
+    "Let's define a final helper function that uses a system prompt.\n",
+    "\n",
+    "> **NOTE:** Try moving the `Make sure you only generate code that is {os_name()}-compatible!` from the user prompt to the system prompt. In our experiments, it seemed to be ignored sometimes when in the system prompt."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 37,
    "metadata": {},
    "outputs": [],
    "source": [
-    "system_prompt = \"You are a helpful software engineer. You write clear, concise, well-commented code.\"\n",
-    "\n",
-    "response = ollama.chat(model='granite-code:8b', messages=[\n",
-    "  {\n",
-    "    'role':'system',\n",
-    "    'content': system_prompt  \n",
-    "  },\n",
-    "  {\n",
-    "    'role': 'user',\n",
-    "    'content': user_prompt\n",
-    "  },\n",
-    "])\n",
-    "print(response['message']['content'])\n"
+    "default_system_prompt = f\"\"\"\n",
+    "    You are a helpful software engineer. You write clear, concise, well-commented code.\"\"\"\n",
+    "\n",
+    "def chat(prompt: str, \n",
+    "         system_prompt:str = default_system_prompt,\n",
+    "         examples: str = examples, \n",
+    "         model: str ='granite-code:8b') -> str:\n",
+    "    user_prompt = f\"\"\"\n",
+    "        {examples}\n",
+    "        Question:\n",
+    "        {prompt} Make sure you only generate code that is {os_name()}-compatible!\n",
+    "        Answer:\"\"\" \n",
+    "    response = ollama.chat(model=model, messages=[\n",
```

Comment:
If we are going to wrap this call, can we highlight it a little more, maybe with spacing around it and a comment above? This is the chunk of code that is doing the work, and I feel that we want it to stand out in the recipe.

Reply:
Done!